src/components/Popup.js

```diff
@@ -30,4 +30,13 @@ export default class Popup {
       .querySelector(".modal__close")
       .addEventListener("click", () => this.close());
   }
+  renderLoading(isLoading) {
```

Comment:
The Popup is a general class, it should not know anything about the form. Please add the method for rendering loading to PopupWithForm and PopupWithConfirmation classes and remove it here

Reply:
done